process_qsos_dr7.m

```diff
@@ -0,0 +1,476 @@
+    this_pixel_mask     =               this_pixel_mask';
+    this_sigma_pixel    =     all_sigma_pixel{quasar_ind};
+    this_sigma_pixel    =               this_sigma_pixel';
+    
+    % convert to QSO rest frame
+    this_rest_wavelengths = emitted_wavelengths(this_wavelengths, z_qso);
+    
+    unmasked_ind = (this_rest_wavelengths >= min_lambda) & ...
+        (this_rest_wavelengths <= max_lambda);% & (this_sigma_pixel>0);
+    % keep complete copy of equally spaced wavelengths for absorption
+    % computation
+    this_unmasked_wavelengths = this_wavelengths(unmasked_ind);
+    
+    % [mask_ind] remove flux pixels with pixel_mask; pixel_mask is defined
+    % in read_spec_dr7.m
+    ind = unmasked_ind & (~this_pixel_mask);% & (this_sigma_pixel>0);
+    
+    this_wavelengths      =      this_wavelengths(ind);
+    this_rest_wavelengths = this_rest_wavelengths(ind);
+    this_flux             =             this_flux(ind);
+    this_noise_variance   =   this_noise_variance(ind);
+    this_sigma_pixel      =      this_sigma_pixel(ind);
+    
+    % c4 existence prior
+    less_ind = (prior.z_qsos < (z_qso + prior_z_qso_increase));
+    this_num_c4    = nnz(prior.c4_ind(less_ind));
+    this_num_quasars = nnz(less_ind);
+    this_p_c4 = this_num_c4 / this_num_quasars;
+    log_priors_c4(quasar_ind) = ...
+        log(                   this_num_c4) - log(this_num_quasars);
+    log_priors_no_c4(quasar_ind) = ...
+        log(this_num_quasars - this_num_c4) - log(this_num_quasars);
+    fprintf('\n');
+    % fprintf(' ...     p(   CIV | z_QSO)  mvn      : %0.3f\n',     this_p_c4);
+    % fprintf(' ...     p(no CIV | z_QSO)        : %0.3f\n', 1 - this_p_c4);
+    
+    % interpolate model onto given wavelengths
+    this_mu = mu_interpolator( this_rest_wavelengths);
+    this_M  =  M_interpolator({this_rest_wavelengths, 1:k});
+    
+   
+    % baseline: probability of no DLA model
+    log_likelihoods_no_c4(quasar_ind) = ...
+        log_mvnpdf_low_rank(this_flux, this_mu, this_M, ...
+        this_noise_variance);
+    
+    log_posteriors_no_c4(quasar_ind) = ...
+        log_priors_no_c4(quasar_ind) + log_likelihoods_no_c4(quasar_ind);
+    
+    % fprintf(' ... log p(D | z_QSO, no CIV)     : %0.2f\n', ...
+        % log_likelihoods_no_c4(quasar_ind));
+    
+    min_z_c4s(quasar_ind) = min_z_c4(this_wavelengths);
+    max_z_c4s(quasar_ind) = max_z_c4(z_qso, max_z_cut);
+    
+    sample_z_c4 = ...
+        min_z_c4s(quasar_ind) +  ...
+        (max_z_c4s(quasar_ind) - min_z_c4s(quasar_ind)) * offset_z_samples;
+
+
+    % Temperature samples
+    
+    % ensure enough pixels are on either side for convolving with
+    % instrument profile
+
+   % building a finer wavelength and mask arrays 
+   % by adding the mean of ith and ith +1 element
+    lenW = length(this_wavelengths);
+        
+    fine_wavelengths = finer(this_wavelengths, nAVG);
+    fine_pixel_mask = finer(this_pixel_mask, nAVG);
```

Comment:
Does `finer` function works for Boolean arrays?

Reply:
It gives an out put of non-Boolean but I manually change it to 0 and 1 so that if it is non-zero I set it to 1.

Reply:
So the output will be a float array, and you convert it to be boolean